library/rsa.c

```diff
@@ -2031,6 +2032,8 @@ static int rsa_rsassa_pkcs1_v15_encode( mbedtls_md_type_t md_alg,
      *                                 TAG-NULL + LEN [ NULL ] ]
      *                 TAG-OCTET + LEN [ HASH ] ]
      */
+    if( 0x08 + oid_size + hashlen >= 0x80 )
```

Comment:
Note: I'm not sure how that is related to the hashlen work. I this is an independent hardening, we should probably backport it?

Reply:
Now that the functions have a clearly defined input length, I wanted to make the code robust to changes that could result in accepting different input lengths (for example, one day this code might run on a long SHAKE output). That meant checking that `hashlen` fits, regardless of any assumption we might make on hashes being small enough.

Indeed I think this change should be backported.

Reply:
The backports don't need to hold up the upcoming release. Filed as a Q3 tech debt item in https://github.com/ARMmbed/mbedtls/issues/4722

Reply:
Hmm
Should this check be backported?
If `md_alg == MBEDTLS_MD_NONE` then we never reach it because we've already returned (in 2019), and if `md_alg != MBEDTLS_MD_NONE` then we have already performed this check in 1974, and none of the values have changed since.
Have I missed something here?

Reply:
I (and the reviewers of this PR) missed that the check was already done earlier. So instead of backporting it, please make a PR on `development` to remove the redundant check!

Reply:
As a matter of fact, I noticed the check was already done earlier but noticed it was inside a conditional, and didn't go to the next step and failed to notice that this path was always taken if we ever get to the point where the new check was added. Should probably have said something, though. Thanks @davidhorstmann-arm for noticing.